### PR TITLE
BigQuery Guide

### PR DESCRIPTION
**Pull request by abidlabs**
Wrote a Guide on how to use Gradio with BigQuery, particularly to highlight the `DataFrame` and `ScatterPlot` component. Some of the credit goes to ChatGPT, but even then it was quite time-consuming so would appreciate any feedback related to SEO so we can maximize its reach @aliabd!

This should be merged after #2806 since it uses it within the code example

### CODE REVIEW

File: guides/05_tabular_data_science_and_plots/using_gradio_with_bigquery.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+We'll be working with the New York Times' COVID dataset that is available on BigQuery. The dataset, named `covid19_nyt.us_counties` contains the latest information about the number of confirmed cases and deaths from COVID across US counties. 
+
+**Prerequisites**: This Guide uses [Gradio Blocks](../01_getting_started/01_quickstart.md#blocks-more-flexibility-and-control), so please familiarize yourself with the Blocks class. 
+
+## Setting up your BigQuery Credentials
+
+To use Gradio with BigQuery, you will need to obtain your BigQuery credentials and use them with the BigQuery Python client. If you already have BigQuery credentials, you can skip this section. If not, you can do this for free in just a couple of minutes:
+
+1. First, log in to your Google Cloud account and go to the Google Cloud Console (https://console.cloud.google.com/)
```
Comment (aliabd) [inline guides/05_tabular_data_science_and_plots/using_gradio_with_bigquery.md:15]: this numbering is incorrect, all bullet points just say 1

File: guides/05_tabular_data_science_and_plots/using_gradio_with_bigquery.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
+# Using Gradio with Google BigQuery
+
+Google BigQuery is a cloud-based big data analytics web service for processing very large data sets. It is a serverless and highly scalable data warehousing solution that enables users to analyze data using SQL-like queries.
+
+In this tutorial, we will show you how to query a BigQuery dataset in Python and display the data *real-time* in a  dashboard using `gradio`.
```
Comment (aliabd) [inline guides/05_tabular_data_science_and_plots/using_gradio_with_bigquery.md:5]: Let's try a summary of key points in the blog here. Something like: 

> How to use Google BigQuery: 
> 1. Setting up Google BigQuery Credentials 
> 2. Using the BigQuery Client 
> 3. Building a Dashboard

File: guides/05_tabular_data_science_and_plots/using_gradio_with_bigquery.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
+# Using Gradio with Google BigQuery
```
Comment (aliabd) [inline guides/05_tabular_data_science_and_plots/using_gradio_with_bigquery.md:1]: Don't like this title, what about something users who don't know about Gradio would search for? Like, "Creating a Dashboard from Google BigQuery", or "Using Google BigQuery from Python"?

Comment (aliabd) [reply guides/05_tabular_data_science_and_plots/using_gradio_with_bigquery.md:1]: Title is most important spot for key words in SEO.

File: guides/05_tabular_data_science_and_plots/using_gradio_with_bigquery.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+Google BigQuery is a cloud-based big data analytics web service for processing very large data sets. It is a serverless and highly scalable data warehousing solution that enables users to analyze data using SQL-like queries.
+
+In this tutorial, we will show you how to query a BigQuery dataset in Python and display the data *real-time* in a  dashboard using `gradio`. 
+
+We'll be working with the New York Times' COVID dataset that is available on BigQuery. The dataset, named `covid19_nyt.us_counties` contains the latest information about the number of confirmed cases and deaths from COVID across US counties.
```
Comment (aliabd) [inline guides/05_tabular_data_science_and_plots/using_gradio_with_bigquery.md:7]: Can you link to the dataset?

File: guides/05_tabular_data_science_and_plots/using_gradio_with_bigquery.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+
+In this tutorial, we will show you how to query a BigQuery dataset in Python and display the data *real-time* in a  dashboard using `gradio`. 
+
+We'll be working with the New York Times' COVID dataset that is available on BigQuery. The dataset, named `covid19_nyt.us_counties` contains the latest information about the number of confirmed cases and deaths from COVID across US counties. 
+
+**Prerequisites**: This Guide uses [Gradio Blocks](../01_getting_started/01_quickstart.md#blocks-more-flexibility-and-control), so please familiarize yourself with the Blocks class. 
+
+## Setting up your BigQuery Credentials
+
+To use Gradio with BigQuery, you will need to obtain your BigQuery credentials and use them with the BigQuery Python client. If you already have BigQuery credentials, you can skip this section. If not, you can do this for free in just a couple of minutes:
```
Comment (aliabd) [inline guides/05_tabular_data_science_and_plots/using_gradio_with_bigquery.md:13]: Link to bigquery python client

File: guides/05_tabular_data_science_and_plots/using_gradio_with_bigquery.md (hunk 3 above)
Comment (aliabd) [reply guides/05_tabular_data_science_and_plots/using_gradio_with_bigquery.md:1]: Key words you should incorporate in the blog post (in section titles, summary, and potentially the title itself): 
* big query (with space)
* bigquery pricing
* google cloud console   
* google cloud platform
* google dataflow

Comment (aliabd) [reply guides/05_tabular_data_science_and_plots/using_gradio_with_bigquery.md:1]: Good, high authority related pages to link to throughout the guide: 
* Chapter 1. What Is Google BigQuery? - O'Reilly
https://www.oreilly.com/library/view/google-bigquery-the/9781492044451/ch01.html
* BigQuery Enterprise Data Warehouse - Google Cloud
https://cloud.google.com/bigquery
* google-cloud-bigquery - PyPI
https://pypi.org/project/google-cloud-bigquery/

File: guides/05_tabular_data_science_and_plots/using_gradio_with_bigquery.md (hunk 5 above)
Comment (aliabd) [reply guides/05_tabular_data_science_and_plots/using_gradio_with_bigquery.md:13]: Can you add a title to this bulleted list, after the description, that's simpler? 

> How to Set Up Google BigQuery: 
> 1. First, log in to .....

File: guides/05_tabular_data_science_and_plots/using_gradio_with_bigquery.md (hunk 2 above)
Comment (abidlabs) [reply guides/05_tabular_data_science_and_plots/using_gradio_with_bigquery.md:5]: Let's do it

File: guides/05_tabular_data_science_and_plots/using_gradio_with_bigquery.md (hunk 5 above)
Comment (abidlabs) [reply guides/05_tabular_data_science_and_plots/using_gradio_with_bigquery.md:13]: Linked the client, but skipped the title to bulleted list as it looked weird

File: guides/05_tabular_data_science_and_plots/using_gradio_with_bigquery.md (hunk 1 above)
Comment (abidlabs) [reply guides/05_tabular_data_science_and_plots/using_gradio_with_bigquery.md]: Fixed

**Comment by abidlabs** [comment]
Thanks for the review @aliabd, address your feedback!

**Pull request merged**